### DISCUSSION TITLE
Adding Avalonia GUI to existing Console .NET App

### DISCUSSION THREAD

**Discussion by salasidis** [AvaloniaUI/Avalonia#15976] [General]
I have an application where I hoped to use a web fr0ont end, but looking to see if a GUI front end with Avalonia may actually be more responsive locally. It would have a data grid, a real time scrolling chart, as well as some local data of the device.

The Console .NET application already exists. Is there an easy or recommended way to add Avalonia to it.

Thanks for any suggestions

**Comment by timunie** [comment]
You should be able to just add an Avalonia project to your sln and run the exe from the console app or the other way around I guess.

**Comment by thevortexcloud** [comment]
Assuming you want to actually have a console app and an Avalonia GUI running side by side: The main thing you need to do is set up Avalonia. Which is fairly straight forwards. Just make `App.axaml` and `App.axaml.cs` files and fill them with what comes with the default template. Then call one of the Avalonia start up methods in the main method.

The winforms interop sample is probably a decent example of the general process of what you would do.

https://github.com/AvaloniaUI/Avalonia/tree/master/samples/interop/WindowsInteropTest

If you don't need them running side by side: Then I would suggest just moving your main application logic into a library project that can be referenced from the console and Avalonia projects. Then make a new Avalonia project that references the library project.

**Comment by salasidis** [comment]
Console displayed in the background - not important (in Raspberry Pi - when started from the rc.local file it does not appear on the desktop).

The Avalonia GUI will be started by this program - presumably as above, and then just work in a kiosk type environment, where user touch inputs can be responded to, but the program will not be able to be closed or stopped.

**Reply by thevortexcloud** [reply]
If your console app is some sort of launcher/bootstrapper/watchdog, making it directly launch an Avalonia executable is probably fine. Although since you are on a Pi, you could probably just use systemd to achieve the same thing depending on what the console app is actually doing.
